addons/web/static/src/js/views/calendar/calendar_controller.js

```diff
@@ -142,6 +172,12 @@ var CalendarController = AbstractController.extend({
         this._updateRecord(event.data);
     },
     /**
+     * Called on 'edit' click action on the dialog generated by _onOpenEvent.
+     */
+    _onEdit: function (open_dialog){
```

Comment:
missing docstring

instead of create a `_onEdit` method, refactor `_onOpenEvent`.
Don't create an `open_dialog` method, check a variable `event.data.editMode` and open the dialog.

click: self._onEdit.bind(self, open_dialog) become something like:

```
click: function () {
    this.trigger_up('openEvent', _.extend({editMode: true}, event.data));
}
```

(It's just so that the code is cleaner and more extensible. You can also remove these changes from the diff)